org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/IndexServiceClientRegressionTests.xtend

```diff
@@ -0,0 +1,57 @@
+package org.testeditor.web.backend.xtext.index
+
+import javax.ws.rs.client.Entity
+import javax.ws.rs.core.Form
+import org.junit.Test
+
+import static javax.ws.rs.core.Response.Status.*
+
+class IndexServiceClientRegressionTests extends AbstractXtextIntegrationTest {
+
+	/**
+	 * Regression test to reproduce an issue in the index service client.
+	 * 
+	 * Observed behavior: in the context of the following request, the index service client tried to serialize the
+	 * Xtext resource's contents to send it as the body of the request to the index service. This seemed to throw off
+	 * the Xtext serializer.
+	 */
+	@Test
+	def void shouldNotCauseXtextSerializationToFail() {
+		// given
+		val tcl = '''
+			package org.testeditor.demo.swing
```

Comment:
tcl should be shortened (most of it is not really relevant for the test)

Reply:
done (in a way that still reproduces the original issue as it did at 12709d2adc1423ef1f4a9bf44159aa96e66a5035)